pkg/apis/apps/validation/validation.go

```diff
@@ -134,6 +134,10 @@ func ValidateStatefulSetSpec(spec *apps.StatefulSetSpec, fldPath *field.Path, op
 		allErrs = append(allErrs, apivalidation.ValidateNonnegativeField(int64(replicaStartOrdinal), fldPath.Child("ordinals.start"))...)
 	}
 
+	if len(spec.ServiceName) > 0 {
```

Comment:
ValidateStatefulSetSpec is also called from ValidateStatefulSetUpdate, so needs to be ratcheting to allow update of existing instances that set this improperly. Add unit tests to exercise this scenario

Reply:
@liggitt updated, should be good.